crates/ruma-client-api/src/push.rs

```diff
@@ -286,9 +286,13 @@ impl PusherIds {
 }
 
 /// Information for an email pusher.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-pub struct EmailPusherData;
+pub struct EmailPusherData {
+    /// Custom data for the pusher.
+    #[serde(flatten, default, skip_serializing_if = "JsonObject::is_empty")]
```

Comment:
I strongly dislike serde's flatten attribute due to how it works internally for deserialization. Can we use `serde(transparent)` instead and revisit if / when we need to add more fields?

Reply:
Done.